straxen/plugins/nveto_recorder.py

```diff
@@ -0,0 +1,352 @@
+            last_interval_seen = np.argwhere(m)[0, 0] + last_interval_seen
+
+            # Now we have to get all associated records and set them to true:
+            for neighbors in record_links:
+                n = 0  # some trial counter
+                ri = ind
+                while n <= 1000:
+                    ri = neighbors[ri]
+                    if ri == -1:
+                        break
+                    else:
+                        result[ri] = True
+
+                if n == 1000:
+                    raise RuntimeWarning('Tried more than 1000 times to find'
+                                         ' neighboring record. This is odd.')
+    return result
+
+
+@export
+def coincidence(records, nfold=4, resolving_time=300):
+    """
+    Checks if n-neighboring events are less apart from each other then
+    the specified resolving time.
+
+    :param records: Can be anything which is of the dtype
+        strax.interval_dtype e.g. records, hits, peaks...
+    :param nfold: coincidence level.
+    :param resolving_time: Time window of the coincidence [ns].
+    :return: array containing the start times and end times of the
+            corresponding intervals.
+
+    Note:
+        The coincidence window is self-extending. If start times of two
+         intervals are exactly resolving_time apart from each other
+         they will be merged into a single interval.
+    """
+    if len(records):
+        start_times = _coincidence(records, nfold, resolving_time)
+        intervals = _merge_intervals(start_times, resolving_time)
+    else:
+        intervals = np.zeros((0, 2), np.int64)
+    return intervals
+
+
+def _coincidence(rr, nfold=4, resolving_time=300):
+    """
+    Function which checks if n-neighboring events are less apart from
+    each other then the specified resolving time.
+
+    Note:
+        1.) For the nVETO recorder we treat every fragment as a single
+            signal.
+        2.) The coincidence window in here is not self extending. Hence
+            we compute only the start times of a coincidence window.
+        3.) By default we cannot test the last n-1 records since we
+            do not know the gap to the next chunk.
+    """
+    # 1. estimate time difference between fragments:
+    start_times = rr['time']
+    mask = np.zeros(len(start_times), dtype=np.bool_)
+    t_diff = np.diff(start_times, prepend=start_times[0])
+
+    # 2. Now we have to check if n-events are within resolving time:
+    #   -> use moving average with size n to accumulate time between n-pulses
+    #   -> check if accumulated time is below resolving time
+
+    # generate kernel:
+    kernel = np.zeros(nfold)
+    kernel[:(nfold - 1)] = 1  # weight last seen by t_diff must be zero since
+    # starting time point e.g. n=4: [0,1,1,1] --> [dt1, dt2, dt3, dt4, ..., dtn]  --> 0*dt1 + dt2 + dt3 + dt4
+
+    t_cum = convolve1d(t_diff, kernel, mode='constant', origin=(nfold - 1) // 2)
```

Comment:
this can be quite slow (I don't know if it is)

Reply:
I know, but at least for the little nveto data I have it was not the slowest part of the processing chain. But maybe this will change once we have all 120 PMTs. So far I tested it with 15 PMTs.